server/controllers/user.controller.js

```diff
@@ -0,0 +1,143 @@
+                status: 400,
+                code: error.code,
+                message: error.message
+            })
+        }
+        return res.status(error.status || 500).send({
+            error: error,
+            message: "An error occured when trying to create the user."
+        })
+    })
+}
+
+exports.findAll = async (req, res) => {
+    hasPermission(res, 3)
+
+    await userService.getAllUsers()
+    .then(response => {
+        return res.send(response)
+    })
+    .catch(error => {
+        return res.status(error.status || 500).send({
+            error: error,
+            message: "An error occured when trying to get all users."
+        })
+    })
+}
+
+exports.find = async (req, res) => {
+    hasPermission(res, 3)
+
+    if (!req.params.id) {
+        return res.status(400).send({
+            message: "Id cannot be empty."
+        })
+    }
+
+    try {
+        let response
+
+        if (req.query.uid)
+            response = await userService.getUserByUID(req.params.id)
+        else
+            response = await userService.getUserById(req.params.id)
+
+        return res.send(response)
+    } catch (error) {
+        return res.status(error.status || 500).send({
+            error: error,
+            message: "An error occured when trying to get the user."
+        })
+    }
+}
+
+exports.modifyUser = async (req, res) => {
+    hasPermission(res, 1)
+
+    if (!req.body 
+        || !req.body.id
+        || !req.body.firebaseUID
+        || !req.body.email
+        || !req.body.steamId
+        || !req.body.role
+        || !req.body.isWhitelisted
+    ) {
+        return res.status(400).send({
+            message: "Content cannot be empty."
+        })
+    }
+
+    const user = {
+        id: req.body.id,
+        firebaseUID: req.body.firebaseUID,
+        email: req.body.email,
+        steamId: req.body.steamId,
+        role: req.body.role,
+        isWhitelisted: req.body.isWhitelisted
+    }
+
+    await userService.updateUser(user)
+    .then(response => {
+        return res.send(response)
+    }) 
+    .catch(error => {
+        return res.status(error.status || 500).send({
+            error: error,
+            message: "An error occured when trying to modify the user."
+        })
+    })
+}
+
+exports.deleteUser = async (req, res) => {
+    hasPermission(res, 1)
+
+    if (!req.params.uid) {
+        return res.status(400).send({
+            message: "firebaseUID cannot be empty."
+        })
+    }
+
+    await userService.deleteUser(req.params.uid)
+    .then(response => {
+        if (response)
+            return res.send({ message: "User deleted successfully" })
+        else
+            return res.status(404).send({ message: `User not found with uid ${req.params.uid}` })
+    })
+    .catch(error => {
+        return res.status(error.status || 500).send({
+            error: error,
+            message: "An error occured when trying to delete the user."
+        })
+    })
+}
```

Comment:
The user with any permission needs to be able to delete his own account if he wants to. Also, make sure the user is removed from the whitelist when the delete user action is called.

Reply:
Will be done in #27